MetaMask/metamask-extension
# Keep ENS name in the URL when redirecting IPFS and IPNS content

This is another take on #7740.

When a user visits "myname.eth" what we are doing right now is:
- Get the contentHash of "myname.eth" from ENS and parse it to get `type` and `hash`.
- If the `type` is "ipfs-ns" (or "ipns-ns"), redirect to `https://[hash].ipfs.gateway` (or `https://[hash].ipns.gateway`).
- There is no trace of "myname.eth" in the redirected URL so the web app has no way to detect it.

With the changes in this PR, we will instead redirect to `https://myname-eth.ipns.gateway/` (for both "ipfs-ns" and "ipns-ns") which will keep the ENS name in the URL. (Resolving `myname.eth` to its actual "ipfs-ns"/"ipns-ns" content will be done by the gateway for us.)

Manual testing steps:  
  - Go to "https://cardpunks.eth/"
  - Observe that the redirected URL is "https://cardpunks-eth.ipns.dweb.link/" and not https://[very long hash].ipfs.dweb.link/
